spec/unit/post_zephir_verifier_spec.rb

```diff
@@ -0,0 +1,443 @@
+        # so expect no warnings
+        verifier.verify_catalog_archive(date: test_date)
+        expect(verifier.errors).to be_empty
+      end
+
+      it "warns if there is a input/output line count mismatch" do
+        # Make a temporary ht_bib_export with just 1 line to trigger error
+        ClimateControl.modify(ZEPHIR_DATA: "/tmp/test/zephir_data") do
+          FileUtils.mkdir_p(ENV["ZEPHIR_DATA"])
+          Zinzout.zout(File.join(ENV["ZEPHIR_DATA"], "ht_bib_export_full_2024-11-30.json.gz")) do |gz|
+            gz.puts "{ \"this file\": \"too short\" }"
+          end
+          # The other unmodified fixtures in CATALOG_ARCHIVE should
+          # no longer have matching line counts, so expect a warning
+          verifier.verify_catalog_archive(date: test_date)
+          expect(verifier.errors.count).to eq 1
+          expect(verifier.errors).to include(/output line count .+ != input line count/)
+        end
+      end
+    end
+
+    describe "#verify_catalog_prep" do
+      test_date = Date.parse("2024-11-30")
+      context "with all the expected files" do
+        it "reports no errors" do
+          # Create and test upd, full, and deletes in @tmpdir/catalog_prep
+          ClimateControl.modify(CATALOG_PREP: @tmpdir) do
+            FileUtils.cp(fixture(File.join("catalog_archive", "zephir_full_20241130_vufind.json.gz")), @tmpdir)
+            FileUtils.cp(fixture(File.join("catalog_archive", "zephir_upd_20241130.json.gz")), @tmpdir)
+            FileUtils.cp(fixture(File.join("catalog_prep", "zephir_upd_20241130_delete.txt.gz")), @tmpdir)
+            verifier = described_class.new
+            verifier.verify_catalog_prep(date: test_date)
+            expect(verifier.errors.count).to eq 0
+          end
+        end
+      end
+
+      context "without any of the expected files" do
+        it "reports an error for each of the three missing files" do
+          ClimateControl.modify(CATALOG_PREP: @tmpdir) do
+            verifier = described_class.new
+            verifier.verify_catalog_prep(date: test_date)
+            expect(verifier.errors.count).to eq 3
+          end
+        end
+      end
+    end
+
+    describe "#verify_dollar_dup" do
+      test_date = Date.parse("2024-12-01")
+      context "with empty file" do
+        it "reports no errors" do
+          ClimateControl.modify(TMPDIR: @tmpdir) do
+            dollar_dup_path = File.join(@tmpdir, "vufind_incremental_2024-12-01_dollar_dup.txt.gz")
+            Zinzout.zout(dollar_dup_path) { |output_gz| }
+            verifier = described_class.new
+            verifier.verify_dollar_dup(date: test_date)
+            expect(verifier.errors).to eq []
+          end
+        end
+      end
+
+      context "with nonempty file" do
+        it "reports one `spurious dollar_dup lines` error" do
+          ClimateControl.modify(TMPDIR: @tmpdir) do
+            dollar_dup_path = File.join(@tmpdir, "vufind_incremental_2024-12-01_dollar_dup.txt.gz")
```

Comment:
Could this one use `expect_not_ok`?

Reply:
esp. if we're passing around Derivative instances that encapsulate filename + date rather than some mix of filenames and dates..